### PR TITLE
fix(batch): fix lookup join lookups a singleton table

### PR DESCRIPTION
**Pull request by chenzl25**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

- As title.

## Checklist For Contributors

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [x] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [x] I have demonstrated that backward compatibility is not broken by breaking changes and created issues to track deprecated features to be removed in the future. (Please refer to the issue)
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Checklist For Reviewers

- [x] I have requested macro/micro-benchmarks as this PR can affect performance substantially, and the results are shown.

## Documentation

- [x] My PR **DOES NOT** contain user-facing changes.

<details><summary>Click here for Documentation</summary>

### Types of user-facing changes

Please keep the types that apply to your changes, and remove the others.

- Installation and deployment
- Connector (sources & sinks)
- SQL commands, functions, and operators
- RisingWave cluster configuration changes
- Other (please specify in the release note below)

### Release note

</details>

#8296

### CODE REVIEW

File: src/frontend/src/optimizer/plan_node/logical_join.rs [hunk 1]
```diff
@@ -502,6 +502,10 @@ impl LogicalJoin {
         let order_col_ids = table_desc.order_column_ids();
         let order_key = table_desc.order_column_indices();
         let dist_key = table_desc.distribution_key.clone();
+        // Avoid to lookup a table with a singleton distribution.
```
Comment (BugenZhao) [inline src/frontend/src/optimizer/plan_node/logical_join.rs:505]: May I ask why we ban this? Sounds feasible technically. 🤔

Comment (chenzl25) [reply src/frontend/src/optimizer/plan_node/logical_join.rs:505]: You are right. The distributed lookup join has already supported it, but the local lookup join failed.

File: src/batch/src/executor/join/local_lookup_join.rs [hunk 2]
```diff
@@ -85,6 +85,12 @@ impl<C: BatchTaskContext> InnerSideExecutorBuilder<C> {
             .iter()
             .map(|&k| k as usize)
             .collect_vec();
+
+        // It is a singleton table.
+        if dist_keys.is_empty() {
+            return Ok(VirtualNode::from_index(0));
```
Comment (BugenZhao) [inline src/batch/src/executor/join/local_lookup_join.rs:91]: What about using `risingwave_storage::table::DEFAULT_VNODE`?

**Review by BugenZhao — APPROVED** [review]
How about also adding an e2e test?

**Pull request merged**